sdktests/server_side_data_system_read_only.go

```diff
@@ -0,0 +1,195 @@
+package sdktests
+
+import (
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/launchdarkly/go-sdk-common/v3/ldreason"
+	"github.com/launchdarkly/go-sdk-common/v3/ldvalue"
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	h "github.com/launchdarkly/sdk-test-harness/v2/framework/helpers"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/ldtest"
+	o "github.com/launchdarkly/sdk-test-harness/v2/framework/opt"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+)
+
+func (s *ServerSideDataSystemTests) doReadOnlyTests(t *ldtest.T) {
+	t.Run("daemon mode", s.daemonModeTests)
+}
+
+func (s *ServerSideDataSystemTests) daemonModeTests(t *ldtest.T) {
+	t.Run("ignores database initialization flag", s.ignoresInitialization)
+	t.Run("can disable cache", s.canDisableCache)
+	t.Run("caches flag for duration", s.cachesFlagForDuration)
+	t.Run("caches flag forever", s.cachesFlagForever)
+}
+
+func (s *ServerSideDataSystemTests) ignoresInitialization(t *ldtest.T) {
+	dataSystem := NewDataSystem()
+	dataSystem.AddPersistence(servicedef.SDKConfigDataSystemPersistence{
+		Store: servicedef.SDKConfigDataSystemPersistenceStore{
+			Type: servicedef.Redis,
+			DSN:  s.persistence.DSN(),
+		},
+		Cache: servicedef.SDKConfigDataSystemPersistenceCache{
+			Mode: servicedef.Off,
+		},
+	})
+	context := ldcontext.New("user-key")
+
+	require.NoError(t, s.persistence.Reset())
+	client := NewSDKClient(t, dataSystem)
+
+	h.RequireEventually(t, func() bool {
+		result := client.EvaluateFlag(t, servicedef.EvaluateFlagParams{
+			FlagKey:      "flag-key",
+			Context:      o.Some(context),
+			ValueType:    servicedef.ValueTypeAny,
+			DefaultValue: ldvalue.String("default"),
+			Detail:       true,
+		})
+
+		return result.Reason.IsDefined() &&
+			result.Reason.Value().GetErrorKind() == ldreason.EvalErrorFlagNotFound
+	}, time.Second, time.Millisecond*20, "flag was found before it should have been")
```

Comment:
Same comment re: pulling into a constant to keep these consistent

Reply:
These parameters are used throughout the code base anywhere the eventually or never methods are used.

Do you want them to be global constants somewhere then? And if so, can I do that in a separate PR so this one doesn't become very noisy with those changes?

Reply:
Oh 😢 

Maybe just go with the flow then.